src/test/java/org/jenkinsci/plugins/workflow/steps/scm/SCMStepTest.java

```diff
@@ -180,6 +180,9 @@ public void scmVars() throws Exception {
 
     @Test public void scmParsesChangelogFileFromFakeChangeLogSCM() {
         rr.then(r -> {
+            // sampleGitRepo is not actually used in this test, but for reasons unclear to me, if you remove the call to
+            // init, checkoutsRestored and gitChangelogSmokes fail when you run the entire test suite.
+            sampleGitRepo.init();
```

Comment:
Any ideas about this? Is this specific to my machine or does it happen for everyone? IDK what caused this issue, my first guess is the parent POM upgrade maybe changed the test execution order so that now `scmParsesChangelogFileFromFakeChangeLogSCM` and `scmParsesUnmodifiedChangelogFile` run first and expose a problem that always existed, but I'm not sure.

The specific error I get is the following:

```
Cloning repository file:///Users/dnusbaum/Projects/workflow-scm-step-plugin/target/tmp/junit4483921317760533160/junit346183201334662032/
 > git init /Users/dnusbaum/Projects/workflow-scm-step-plugin/target/tmp/j h6357427053435370264/workspace/p # timeout=10
ERROR: Error cloning remote repo 'origin'
hudson.plugins.git.GitException: Could not init /Users/dnusbaum/Projects/workflow-scm-step-plugin/target/tmp/j h6357427053435370264/workspace/p
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$5.execute(CliGitAPIImpl.java:997)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$2.execute(CliGitAPIImpl.java:752)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1224)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1302)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:125)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:93)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:80)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: hudson.plugins.git.GitException: Error performing git command: git init /Users/dnusbaum/Projects/workflow-scm-step-plugin/target/tmp/j h6357427053435370264/workspace/p
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:2457)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:2380)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:2376)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommand(CliGitAPIImpl.java:1923)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$5.execute(CliGitAPIImpl.java:995)
	... 12 more
Caused by: java.lang.IllegalThreadStateException
	at java.lang.ThreadGroup.addUnstarted(ThreadGroup.java:867)
	at java.lang.Thread.init(Thread.java:405)
	at java.lang.Thread.init(Thread.java:349)
	at java.lang.Thread.<init>(Thread.java:678)
	at java.util.concurrent.Executors$DefaultThreadFactory.newThread(Executors.java:613)
	at hudson.util.DaemonThreadFactory.newThread(DaemonThreadFactory.java:46)
	at hudson.util.ClassLoaderSanityThreadFactory.newThread(ClassLoaderSanityThreadFactory.java:23)
	at hudson.util.NamingThreadFactory.newThread(NamingThreadFactory.java:52)
	at hudson.util.ExceptionCatchingThreadFactory.newThread(ExceptionCatchingThreadFactory.java:50)
	at java.util.concurrent.ThreadPoolExecutor$Worker.<init>(ThreadPoolExecutor.java:619)
	at java.util.concurrent.ThreadPoolExecutor.addWorker(ThreadPoolExecutor.java:932)
	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1378)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
	at hudson.Proc.joinWithTimeout(Proc.java:159)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:2443)
	... 16 more
```

I looked through `GitSampleRepoRule`, `AbstractSampleDVCSRepoRule`, and `AbstractSampleRepoRule`, but nothing really stood out to me.

Reply:
Definitely not just your machine, I ran into the same issues for the entire test suite without the line and got the same error. It does feel like the test order is getting messed up

Reply:
I don't see the same issue.  The tests pass on my Linux machine with JDK 8 and JDK 11.  The test pass on my Windows machine with JDK 8.

Reply:
Even if you remove `sampleGitRepo.init();` on line 185?

Reply:
Pushed b252008 just to see what happens in the CI build.

Reply:
Weird, so b252008 did pass, maybe it's related to my `git` version? I am running 2.26.2 which is from April 2020, but it looks like there have been around 7 releases since then. I'll try updating.

Reply:
That didn't help either. I thought maybe it could be related to https://github.com/awslabs/git-secrets, which I had activated in my local repo due to employer requirements, but I deactivated it and still ran into the same issue. At this point I'm not really sure what is happening. I guess it would be interesting to know in what order the tests run for users who are not seeing any problems to see if the difference is just the test ordering, but I think [this configuration](https://github.com/jenkinsci/plugin-pom/blob/09b5b45beb41bfe9455413ea3a5255b5672454af/pom.xml#L742) is supposed to make that consistent for everyone.

Reply:
When I remove the `sampleGitRepo.init()` it fails on Java 11 but works on Java 8.

Reply:
In 4b25eb9 I added a JUnit annotation so that the test methods are always executed in alphabetical order. That should at least make the results consistent in all environments even if we don't currently understand _why_ the tests fail if they run in a different order.